lib/travis/build/addons/rethinkdb.rb

```diff
@@ -0,0 +1,44 @@
+require 'shellwords'
+require 'travis/build/addons/base'
+
+module Travis
+  module Build
+    class Addons
+      class Rethinkdb < Base
+        SUPER_USER_SAFE = true
+
+        RETHINKDB_GPG_KEY = '0x3A8F2399'
+
+        def after_prepare
+          sh.fold 'rethinkdb' do
+            sh.if "$(uname) != 'Linux'" do
+              sh.echo "The RethinkDB addon only works on Linux.", ansi: :red
+            end
+            sh.else do
+              sh.echo "Installing RethinkDB version #{rethinkdb_version}", ansi: :yellow
+              sh.cmd "service rethinkdb stop", sudo: true
+              sh.cmd "apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 #{RETHINKDB_GPG_KEY}", sudo: true
+              sh.cmd 'echo -e "\ndeb http://download.rethinkdb.com/apt $(lsb_release -cs) main" >> /etc/apt/sources.list', sudo: true
```

Comment:
This doesn't work, because `sudo` cannot redirect output this way.

```ruby
sh.cmd 'echo -e "\ndeb http://download.rethinkdb.com/apt $(lsb_release -cs) main" | sudo tee -a /etc/apt/sources.list >/dev/null'
```